EMS/EMS.ConsoleManager.cs

```diff
@@ -0,0 +1,456 @@
+    {
+        PrintEmployeesTableHeader();
+        foreach (EmployeeDetails employee in employees)
+        {
+            string fullName = $"{employee.FirstName ?? null} {employee.LastName ?? null}";
+            string dob = employee.Dob?.ToString("dd-MM-yyyy") ?? null;
+            string email = employee.Email ?? null;
+            string mobileNumber = employee.MobileNumber ?? null;
+            string locationName = employee.LocationName ?? null;
+            string jobTitle = employee.RoleName ?? null;
+            string departmentName = employee.DepartmentName ?? null;
+
+            Console.WriteLine($" {employee.EmpNo}\t\t|{fullName,-20}\t|{employee.StatusName,-10}\t|{dob}\t|{email,-30}\t|{mobileNumber}\t|{locationName,-10}\t\t|{jobTitle,-30}\t|{departmentName}");
+        }
+        Console.WriteLine("-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------\n");
+    }
+
+    private static void PrintEmployeesTableHeader()
+    {
+        Console.WriteLine("\nEmployee Details:\n");
+        Console.WriteLine("-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------");
+        Console.WriteLine(" Employee ID\t|Name\t\t\t|Status\t\t|Date of Birth\t|Email\t\t\t\t|Mobile Number\t|Location\t\t|Job Title\t\t\t|Department");
+        Console.WriteLine("-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------");
+    }
+
+    // Roles related partial functions
+    private static partial Role GetRoleDataFromConsole()
+    {
+        PrintConsoleMessage("Enter Role details:\n");
+        bool required = true;
+
+        string roleFilePath = GetIConfiguration()["RoleJsonPath"];
+        List<Role> roles = _jsonUtils.ReadJSON<Role>(roleFilePath);
+
+        Role role = new()
+        {
+            Id = roles.Count + 1,
+            DepartmentId = GetIdFromUser<Department>("Department", "DepartmentJsonPath", required),
+            RoleName = GetDataFromField("Role Name", required)!
+        };
+
+        return role;
+    }
+
+    private static void PrintRolesTableHeader()
+    {
+        Console.WriteLine("\nRoles Details:\n");
+        Console.WriteLine("------------------------------------------------------------------------------------");
+        Console.WriteLine(" Role ID\t|Role Name\t\t\t|Department Name");
+        Console.WriteLine("------------------------------------------------------------------------------------");
+    }
+
+    private static partial void PrintRoles(List<Role> roles)
+    {
+        Dictionary<int, string>? departmentNames = _dropdownDal.GetDepartments();
```

Comment:
application layer shouldn't interact with DAL directly. It should interact with BAL.